api-channel/src/main/java/com/synopsys/integration/alert/api/channel/CommonChannelDistributionValidator.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * api-channel
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.alert.api.channel;
+
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.common.descriptor.config.ui.ChannelDistributionUIConfig;
+import com.synopsys.integration.alert.common.descriptor.validator.ConfigurationFieldValidator;
+
+@Component
+public class CommonChannelDistributionValidator {
+    public void validate(ConfigurationFieldValidator configurationFieldValidator) {
+        configurationFieldValidator.validateRequiredFieldIsNotBlank(ChannelDistributionUIConfig.KEY_CHANNEL_NAME);
+        configurationFieldValidator.validateRequiredFieldIsNotBlank(ChannelDistributionUIConfig.KEY_NAME);
+        configurationFieldValidator.validateRequiredFieldIsNotBlank(ChannelDistributionUIConfig.KEY_FREQUENCY);
```

Comment:
Since this is a static list of options, we should validate that the selected item is in the list of options (Don't need this for endpoint select since it's dynamic data).

Reply:
Validated that Frequency is one of the possible options!